### DISCUSSION TITLE
Useful links

### DISCUSSION THREAD

**Discussion by eeholmes** [UW-Upwelling-Project/Summer-2023#5] [General]
Data

* EarthData https://search.earthdata.nasa.gov/
* PODAAC (oceanography) https://podaac.jpl.nasa.gov/cloud-datasets/dataaccess
* Coastwatch ERDDAP https://coastwatch.noaa.gov/cwn/data-access-tools/erddap-noaa-coastwatch.html
* NOAA World Ocean Database https://www.ncei.noaa.gov/products/world-ocean-database

Data Access Tutorials

* link

References on Indian Ocean upwelling

* link
*

**Comment by minhphan03** [comment]
[Automated detection of coastal upwelling in the Western Indian Ocean: Towards an operational “Upwelling Watch” system](https://www.frontiersin.org/articles/10.3389/fmars.2022.950733/full)

**Comment by minhphan03** [comment]
[Upwelling and Downwelling and association with surface currents/waves](https://rwu.pressbooks.pub/webboceanography/chapter/9-5-currents-upwelling-and-downwelling/)

**Comment by NaNa7Miiii** [comment]
## Benchmark Pretrained Models
| Model Name             | Input size | Top 1 Accuracy (%) | Size (MB) | # of Parameters (M) | FLOPs (G) | # of Activations (million) | Latency on CPU (ms) | Latency on ANE (ms) | Latency on GPU (ms) | Cpu/Gpu Operation Ratio | Model Link                                                   |
|------------------------|------------|--------------------|-----------|---------------------|-----------|----------------------------|---------------------|---------------------|---------------------|-------------------------|--------------------------------------------------------------|
| Nextvit_small          | 224        | 82.5               | 126.98    | 31.763              | 5.811     | 18.4387                    | 35.2382             | 3.5006              | 129.4218            | 85/167                  | [Link](https://github.com/bytedance/Next-ViT)                 |
| Nextvit_base           | 224        | 83.2               | 179.15    | 44.819              | 8.294     | 23.7119                    | 49.7409             | 4.4305              | 193.8774            | 129/245                 | [Link](https://github.com/bytedance/Next-ViT)                 |
| Nextvit_large          | 224        | 83.6               | 231.33    | 57.874              | 10.777    | 28.99                      | 66.3024             | 5.2163              | 259.4616            | 173/323                 | [Link](https://github.com/bytedance/Next-ViT)                 |
| Efficientnet_b0        | 224        | 77.692             | 17.3      | 5.289               | 0.422     | 6.75                       | 16.9126             | 1.7491              | 26.7446             | 234/65                  | [Link](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_b0.html) |
| Efficientnet_b1        | 240        | 78.642             | 27.5      | 7.794               | 0.62      | 9.36                       | 23.3418             | 2.5068              | 36.3961             | 326/102                 | [Link](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_b1.html) |
| Efficientnet_b2        | 260        | 80.608             | 32.5      | 9.11                | 0.712     | 9.81                       | 36.6151             | 3.6823              | 42.6457             | 223/219                 | [Link](https://pytorch.org/vision/master/models/generated/torchvision.models.efficientnet_b2.html) |
| Efficientnet_b3        | 300        | 82.008             | 45        | 12.233              | 1.033     | 13.01                      | 65.1021             | 5.5076              | 47.3737             | 76/444                  | [Link](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_b3.html) |
| Efficientnet_b4        | 380        | 83.384             | 73.7      | 19.342              | 1.596     | 17.05                      | 111.996             | 10.374              | 55.6502             | 66/576                  | [Link](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_b4.html) |
| Efficientnet_b5        | 456        | 83.444             | 118.4     | 30.39               | 2.483     | 23.42                      | 262.8067            | 21.99               | 110.4733            | 80/699                  | [Link](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_b5.html) |
| Efficientnet_b6        | 528        | 84.008             | 169.6     | 43.041              | 3.522     | 29.97                      | 490.5762            | 52.0414             | 201.4061            | 92/807                  | [Link](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_b6.html) |
| Efficientnet_b7        | 600        | 84.122             | 263.7     | 66.348              | 5.387     | 40.09                      | 947.5508            | 86.9574             | 364.3727            | 112/984                 | [Link](https://pytorch.org/vision/main/models/generated/torchvision.models.efficientnet_b7.html) |
| Efficientnet_v2_s      | 224        | 84.228             | 85.6      | 21.458              | 2.878     | 12.19                      | 35.8593             | 3.0237              | 48.5658             | 2/501                   | [Link](https://pytorch.org/vision/main/models/efficientnetv2.html) |
| Efficientnet_v2_m      | 224        | 85.112             | 216       | 54.139              | 5.407     | 19.62                      | 70.4114             | 5.3902              | 61.1385             | 2/724                   | [Link](https://pytorch.org/vision/main/models/efficientnetv2.html) |
| Efficientnet_v2_l      | 224        | 85.808             | 473.1     | 119                 | 12.31     | 34.5                       | 126.4621            | 10.8362             | 89.462              | 2/1003                  | [Link](https://pytorch.org/vision/main/models/efficientnetv2.html) |
| Efficientformer_v1_L1  | 224        | 80.2               | 49.1      | 12.29               | 1.302     | 5.53                       | 92.7071             | 1.5094              | 34.8053             | 42/148                  | [Link](https://github.com/snap-research/EfficientFormer)      |
| Efficientformer_v1_L3  | 224        | 82.4               | 125.7     | 31.406              | 3.929     | 12                         | 208.3578            | 2.8547              | 60.2183             | 141/228                 | [Link](https://github.com/snap-research/EfficientFormer)      |
| Efficientformer_v1_L7  | 224        | 83.3               | 329.2     | 82.229              | 10.166    | 24.45                      | 423.6137            | 8.2246              | 89.8642             | 273/308                 | [Link](https://github.com/snap-research/EfficientFormer)      |
| Efficientformer_v2_S0  | 224        | 76.2               | 14.2      | 3.6                 | 0.407     | 5.3                        | 51.3086             | 0.9832              | 23.0806             | 107/121                 | [Link](https://github.com/snap-research/EfficientFormer)      |
| Efficientformer_v2_S1  | 224        | 79.7               | 24.5      | 6.186               | 0.668     | 7.66                       | 77.493              | 1.1349              | 46.6044             | 57/217                  | [Link](https://github.com/snap-research/EfficientFormer)      |
| Efficientformer_v2_S2  | 224        | 82                 | 51        | 12.71               | 1.272     | 11.77                      | 125.7969            | 2.0281              | 72.8974             | 76/350                  | [Link](https://github.com/snap-research/EfficientFormer)      |
| Efficientformer_v2_L   | 224        | 83.5               | 109.14    | 26.322              | 2.952     | 18.54                      | 189.1436            | 2.9518              | 111.8996            | 108/468                 | [Link](https://github.com/snap-research/EfficientFormer)      |

For the calculation of flops, parameters, and activations, I use the [fvcore](https://github.com/facebookresearch/fvcore/blob/main/docs/flop_count.md) library.

**Comment by minhphan03** [comment]
[ The Impact of Geostrophic Transport on the Temporal and Spatial Structure of Wind-Driven Coastal Upwelling/Downwelling over the Persian Gulf ](https://www.mdpi.com/2073-4441/15/6/1168#B30-water-15-01168)
This paper went through models and methods used on analyzing wind speeds... maybe we can implement one or two?

**Comment by minhphan03** [comment]
[Wind_driven_coastal_upwelling_along_the(1).pdf](https://github.com/UW-Upwelling-Project/Summer-2023/files/11879714/Wind_driven_coastal_upwelling_along_the.1.pdf)

Abstract: Using the long-term wind time series, coastal upwelling was investigated in the northern Persian Gulf. The shoreline was divided into 34 segments from NW to SE and the **daily Ekman transport components were computed**. Based on the Ekman transport components, a coastal upwelling index was calculated... To investigate the results, sea surface temperature (SST) and sea level anomalies (SLAs) were analysed.

**Comment by NaNa7Miiii** [comment]
Time Record Idea:
 ```
// Run the actual inference (500 iterations)
long startTime = System.currentTimeMillis();
for (int i = 0; i < 500; i++) {
    session.run(Collections.singletonMap("input.1", inputTensor));
}
long endTime = System.currentTimeMillis();
long inferenceTime = (endTime - startTime) / 500; // Average inference time in ms
```
